### DISCUSSION TITLE
Accessing parent scope in custom wrapper/template?

### DISCUSSION THREAD

**Discussion by koolkunz** [ngx-formly/ngx-formly#2836] [Q&A]
I have recently started work on shifting my application from angularjs with formly to angular10 with formly.

In angularjs I could access the parent scope from the custom wrappers js file simply by using $scope.$parent. What is the recommended way to achieve that in angular10 with formly?

if you see the code I have linked, what I want is that on the click "Test" button which is inside a custom wrapper, I want to call the function ( testclickfromwrapper ) in the main app.component.ts file.

https://stackblitz.com/edit/angular-gjleoz-tjmlmw?file=src/app/panel-wrapper.component.ts

Any help would be appreciated!

**Comment by aitboudad** [comment]
You can get the parent field through `parent` property: `this.field.parent`.
For your use case, you may try using `formState` instead: https://stackblitz.com/edit/angular-gjleoz-hqyxhh
